### ISSUE TITLE
Malicious node(s) strategy

### ISSUE THREAD

**Issue by piotr-dziubecki** [casper-network/roadmap#19]
Some scenarios benefit from a malicious node exhibiting one or more behaviors that the network should be resilient against. We currently lack a creditable strategy to create and support such malicious nodes. Designing and maintaining such nodes incurs a significant cost over time.

**Comment by EdHastingsCasperAssociation** [comment]
@Fraser999 did anything useful come out of your conversation w/ Andreas on this last week?

**Comment by Fraser999** [comment]
@EdHastingsCasperLabs Yes, we discussed some low hanging fruit which could be tested, perhaps using a minimal client as discussed previously.  These would include things like trying to provide panoramas with an invalid number of validators.

I was advised to look through what is being checked in validate and pre-validate functions in state.rs, both to seek ways to exercise them and to draw inspiration for checks which might have been missed there.

However, one point which Andreas, Bart and I seemed to be aligned on was that we think to test anything other than the most basic of attacks, the malicious node will likely need to be running an actual consensus component, with all the necessary supporting components unfortunately.

That being said, I'll still approach this with a view to trying to avoid that for as long as possible.
